rts/Lua/LuaUtils.cpp

```diff
@@ -1798,3 +1799,45 @@ void LuaUtils::PushAttackerInfo(lua_State* L, const CUnit* const attacker)
 	lua_pushnil(L);
 }
 #endif
+
+
+void LuaUtils::TracyRemoveAlsoExtras(char* script)
+{
+	// tracy's built-in remover; does not handle our local TracyExtra functions
+	tracy::LuaRemove(script);
+
+#ifndef TRACY_ENABLE
+	// Our extras are handled manually below, the same way Tracy does.
+	// Code is on BSD-3 licence, (c) 2017 Bartosz Taudul aka wolfpld
+
+	const auto FindEnd = [] (char *ptr) {
+		unsigned int cnt = 1;
+		while (cnt) {
+			     if (*ptr == '(') ++ cnt;
+			else if (*ptr == ')') -- cnt;
+			++ ptr;
+		}
+		return ptr;
+	};
+
+	const auto Wipe = [&script, FindEnd] (size_t offset) {
+		const auto end = FindEnd(script + offset);
+		memset(script, ' ', end - script);
+		script = end;
+	};
+
+	while (*script) {
+		if (strncmp(script, "tracy.LuaTracyPlot", 18)) {
+			++ script;
+			continue;
+		}
+
+		if (!strncmp(script + 18, "Config(", 7))
+			Wipe(18 + 7);
+		else if (!strncmp(script + 18, "(", 1))
```

Comment:
The 18, 7 and 1 are magic but it should be obvious from context what they are, and named constants would be pretty unwieldy.

Reply:
obvious from context? what are those numbers, why not a little comment?

Reply:
I could do it like this but it looks bad imo (the strings look very similar and would be used multiple times)
```suggestion
			Wipe("tracy.LuaTracyPlotConfig("sv.size());
		else if (!strncmp(script + "tracy.LuaTracyPlot"sv.size(), "(", "("sv.size())
```

Reply:
since they get used several times then getting those in a constexpr earlier could do the job and be more understandable, but as you prefer.

Reply:
I thought about that but I can't figure out good var names.

Reply:
classic problem XD

what about:

```
constexpr auto plotConfigStr = "tracy.LuaTracyPlotConfig("
constexpr auto plotStr = "tracy.LuaTracyPlot"
constexpr auto plotConfigStrSize = ... 
constexpr auto plotStrSize = ...
```

for the parenthesis size I think just leave the 1 XD too much hassle, otherwise a comment saying smth like "the numbers are sizes of strings here" or so could already do a lot to make it more "intuitive".

Reply:
Added a comment.